libnavigation-core/src/main/java/com/mapbox/navigation/core/arrival/ArrivalProgress.kt

```diff
@@ -0,0 +1,25 @@
+package com.mapbox.navigation.core.arrival
+
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.navigation.base.trip.model.RouteLegProgress
+import com.mapbox.navigation.base.trip.model.RouteProgress
+
+/**
+ * This class helps digest [RouteProgress] data specifically for arrival experiences.
+ *
+ * @param arrivalOptions the options used for arrival [ArrivalOptions]
+ * @param hasMoreLegs true when approaching a waypoint, false when approaching
+ *     the final destination
+ * @param coordinateIndex the index for looking up [RouteOptions.coordinates],
+ *  [RouteOptions.radiuses], [RouteOptions.bearings], [RouteOptions.approaches],
+ *  [RouteOptions.waypointNames]
+ * @param routeProgress the current [RouteProgress]
+ * @param routeLegProgress the current [RouteLegProgress]
+ */
+class ArrivalProgress internal constructor(
+    val arrivalOptions: ArrivalOptions,
+    val hasMoreLegs: Boolean,
+    val coordinateIndex: Int,
+    val routeProgress: RouteProgress,
+    val routeLegProgress: RouteLegProgress
```

Comment:
This entire object comes from `RouteProgress`, but looking to organize it into relevant objects. Added this one because it's nullable from `RouteProgress`.

It may be best to iterate on the definition of this object, possibly bring in nav-native.

Reply:
Starting to add an "SDK arrival api" section to the building highlight spec
https://docs.google.com/document/d/16yRtY_U-0fGSedg1heEu4TaIWgTnNov4HZX84bG270U/edit